### PR TITLE
Remove wpilib.run

### LINKED ISSUE #51: RFC: remove `python robot.py xxx`
In pursuit of [making installation simpler](https://github.com/robotpy/robotpy-installer/issues/70#issuecomment-1865427265), I'm contemplating changing the way that users launch their code (which has been the same since the 2015 season). It is currently a week before kickoff, but most of this is fairly simple.

For example, to run simulation:

---

## Currently

```
py robot.py sim          # Windows
python robot.py sim      # Linux/macOS
./robot.py sim           # Linux/macOS w/executable bit set
```

And users must also add the following magic to their file:

```python
if __name__ == "__main__":
    wpilib.run(MyRobot)
```
---
## Proposed

Users no longer need to add the magic to their file. The run command will assume that the robot is in `robot.py`, and that it is a subclass of `wpilib.RobotBase` and it is defined in `robot.py`. If there are multiple robot classes, the user will need to add a setting to `pyproject.toml`.

```
robotpy sim              # Installed python script in path (windows or linux)
py -m robotpy sim        # Windows
python -m robotpy sim    # Linux/macOS
```

As an alternative (could do the above and this), a user could also create a file `__main__.py`

```python
import robotpy.launcher
```

And then you could do:

```
py . sim          # Windows
python . sim      # Linux/macOS
./__main__.py sim # Linux/macOS w/executable bit set
```

---

As part of this, I propose removing the `robotpy-installer` command, and move all of its installation actions into the `robotpy` command via the extension mechanism that we currently use to register new subcommands of `robot.py`.

```
robotpy download
robotpy install
.. etc
```

In particular, this solves the chicken/egg problem I ran into when contemplating how to implement a `sync` command that a user could use to ensure their local robotpy installation is up to date. If a user has an import that isn't resolvable, then `robot.py sync` isn't possible to implement, so users would need to learn two commands. 

---

Existing users have a lot of muscle memory that will need to be changed, but I think it would be better to do this change before the first official season instead of afterwards.

Not requiring robot.py to be ran directly will allow us to create better error messages for users with missing imports, make some aspects of unit tests simpler to change in the future, and provide a unified entrypoint for users to interact with robotpy. 

---

## Questions

* Is this too risky to do a week before kickoff? 
   * Doing this isn't technically challenging, just have to shuffle a lot of pieces around
   * ... haven't written any installation docs for frc-docs, so no risk that it doesn't get changed :-D
* Do the advantages of a unified launcher outweigh destroying existing user's muscle memory?
   * Existing documentation on the internet says do `python robot.py XXX`

### PR DESCRIPTION
**Pull request by virtuald**
- Corresponding logic will be moved to robotpy-meta
- __main__ will no longer be the user's robot code, but I don't think anything uses that?
- Fixes #51

### CODE REVIEW

File: subprojects/robotpy-wpilib/wpilib/_impl/start.py [hunk 1]
```diff
@@ -1,12 +1,85 @@
...
+                data.get("git-desc", "<unknown>"),
+                data.get("git-branch", "<unknown>"),
+            )
+
+    logger = logging.getLogger("wpilib")
+
+    try:
+        import robotpy.version
+    except ImportError:
+        robotpy_version = None
```
Comment (auscompgeek) [inline subprojects/robotpy-wpilib/wpilib/_impl/start.py:51]: I know this just moved files, but it looks like this variable is unused?
```suggestion
        pass
```

File: subprojects/robotpy-wpilib/wpilib/_impl/start.py [hunk 2]
```diff
@@ -1,12 +1,85 @@
...
+        logger.info("RobotPy version %s", robotpy.version.__version__)
+
+    logger.info("WPILib version %s", wpilib.__version__)
+
+    if wpilib.RobotBase.isSimulation():
+        logger.info("Running with simulated HAL.")
+
+        # check to see if we're on a RoboRIO
+        # NOTE: may have false positives, but it should work well enough
+        if os.path.exists("/etc/natinst/share/scs_imagemetadata.ini"):
```
Comment (auscompgeek) [inline subprojects/robotpy-wpilib/wpilib/_impl/start.py:62]: I do hope this is unreachable these days 😅

(I suppose someone could forcibly install the raspbian sim HAL wheel somehow?)

File: subprojects/robotpy-wpilib/wpilib/src/rpy/Filesystem.inc [hunk 3]
```diff
@@ -10,11 +10,20 @@ static fs::path getMainPath() {
   py::gil_scoped_acquire gil;
   py::dict locals;
   py::exec(R"(
```
Comment (auscompgeek) [inline subprojects/robotpy-wpilib/wpilib/src/rpy/Filesystem.inc:12]: Would it be worth defining this in pure Python instead? This is a fair bit of embedded Python code now.

**Review by auscompgeek — APPROVED** [review]

File: subprojects/robotpy-wpilib/wpilib/src/rpy/Filesystem.inc (hunk 3 above)
Comment (virtuald) [reply subprojects/robotpy-wpilib/wpilib/src/rpy/Filesystem.inc:12]: This was used by some vendor dep I think?

**Pull request merged**